stdlib/stdlib.ml

```diff
@@ -540,7 +540,10 @@ let at_exit f =
   let g = !exit_function in
   exit_function := (fun () -> f(); g())
 
-let do_at_exit () = (!exit_function) ()
+let do_at_exit () =
+  (!exit_function) ();
+  (* MPR#7796: make sure these at-exit functions will not be run again *)
+  exit_function := (fun () -> ())
```

Comment:
Can it happen that `!exit_function` runs `at_exit` several times and then `do_at_exit` again? In this case, with your definition, I think you will get repetitions, while

```ocaml
  let f = !exit_function in
  exit_function := (fun () -> ());
  f ()
```

does not.